packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useMemo } from 'react';
+
+import {
+  TableHeader,
+  Row,
+  Cell,
+  cx,
+  IndexKeysBadge,
+} from '@mongodb-js/compass-components';
+
+import TypeField from './type-field';
+import SizeField from './size-field';
+import UsageField from './usage-field';
+import PropertyField from './property-field';
+import IndexActions from './index-actions';
+
+import {
+  rowStyles,
+  indexActionsCellStyles,
+  tableHeaderStyles,
+  cellStyles,
+  nestedRowCellStyles,
```

Comment:
Not really a pattern we use often and leaks a bit more implementation detail than probably needed. Do you think it would be possible to hide this behind the interface of the table, so that it's still the one responsible for rendering rows and cells?

Reply:
I tried that at first but I found it awkward without the Row and Cell props exported by leafygreen. I suppose I can make them dumb and only accept a few props..

Reply:
I tried. Weird stuff happening in the Row component that I can't figure out so I reverted it again for now. I think we can try again in a follow-up and I'd like to move on for now and unblock people.